### ISSUE TITLE
allow listening for dotfile changes when using -d

### ISSUE THREAD

**Issue by woodgiraffe** [eradman/entr#77]
Hi,

it seems that when using directory-watches via '-d' dotfiles are always ignored, however my use-case would include also watching for additions of dotfiles, I'm curious if there is a way to sidestep this behavior atm and if not, if this could be added in a future release or there are issues that would prevent this from working.

Thanks for your consideration and feedback!

**Comment by eradman** [comment]
The intent of ignoring dotfiles in directories monitored with the `-d` flag was to prevent editors from triggering a directory change if you merely opened a file. Suppose a directory contains a dotfile:

```
proj/.gitignore
```

You can watch for changes to individual dot-files by listing them explicitly

```
ls -d proj/.gitignore | entr -d echo "file changed" /_
```

But running `vim .gitignore` won't trigger an event even though vim writes a new scratch file to the directory. Can you add some detail about your use case?

One possible solution is to monitor dot-files by default, but keep a list of file extensions to ignore (`.*.swp` for example)

**Comment by woodgiraffe** [comment]
Thank you for the response, thanks for consideration and thanks for this software!

My use case is a debug-script in which a directory is recursively monitored and any files added or changed are automatically committed and pushed to a bare git repository, which is then mounted in a container. It's dirty, but surprisingly useful as it helps me iterate faster to locally debug/explore gitops related projects that force me through this workflow. Other than this dotfile related problem, it's been surprisingly robust as well too, even tho I hacked it together rather quickly.

The `entr` call looks like this currently:

```
        info "watching $in => $out"
        while true; do
            find . -type d > "$tmpfile"
            find . -name "*" >> "$tmpfile"
            find . -name "*.*" >> "$tmpfile" # dotfiles
            sed -i '/.git/d' "$tmpfile"
            
            # remove duplicates
            cp "${tmpfile}" "${tmpfile}_tmp"
            sort -u "${tmpfile}_tmp" > "$tmpfile"
            rm "${tmpfile}_tmp"

            cat "$tmpfile" \
                | entr -pds "\"$0\" \"--autocommit\" \"$in\" \"$out\""
        done
```

As you have described, this picks up dotfiles only when they have been found during startup, by being explicitly listed here via `find` and then passed to `entr`'s stdin.
If dotfiles are however added to a watched directory during runtime, they are ignored.

- - -

I really like the idea of exposing a list of ignore glob-patterns for directory changes!

This could potentially also default to the current "*.*" implementation so that it doesn't break scripts that rely on this behavior?
The shortopt could be named "-di" ("directory ignore"), use a default if never used, and builds a list of ignore glob-patterns when passed multiple times?

Just some ideas about the interface in case you consider this a worthwhile addition/change.


Thanks for your time on this matter!

**Comment by eradman** [comment]
I finally put together a patch using fnmatch(3) to ignore directories based on a glob pattern, but globs don't have an `or` operator or anything similar. Also it seemed odd to set the pattern to an empty string in order to disable the behavior.

Instead would it be enough to simply disable the check for filenames that start with `.`?

**Comment by eradman** [comment]
In commit 09ec07b I tweaked the `-d` flag so that if you specify it more than once all changes are recognized.

**Comment by eradman** [comment]
This feature was included in the 5.1 relesae
